docs/design/backups.md

```diff
@@ -0,0 +1,83 @@
+# Backup & Restore
+
+In particularly disastrous situations, we would like to be able to revert the etcd cluster to a previously known healthy state.
+This document lays out the design of how we want the healthy state to be captured, securely stored & safely restored to operator-managed etcd clusters.
+
+## Backup Resource
+
+We will introduce a new custom resource: `EtcdBackup`.
+
+```yaml
+apiVersion: etcd.improbable.io/v1alpha1
+kind: EtcdBackup
+metadata:
+  name: my-cluster-backup
+  namespace: default
+spec:
+  type: snapshot
+  destination:
+    gcsBucket:
+      bucketName: my-cluster-backup-bucket
+      creds:
+        secretRef: 
+          name: my-cluster-backup-bucket-credentials
+          namespace: default
+```
+
+This resource specifies that we desire a backup to be immediately taken as a [snapshot](https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/recovery.md#snapshotting-the-keyspace) from the etcd API.
+This snapshot will be saved to a GCS bucket, accessed with credentials read from a given secret.
+The file would be named with a timestamp when the backup was started, for uniqueness.
```

Comment:
nit, and might be okay to ignore.. but how do we ensure two backups don't start at the same time? Just process one at a time in the controller?

Reply:
your comment made me worry that we wouldn't be able to update any clusters/peers while a backup is being taken, as kubebuilder does not allow concurrent reconciles. but from a quick grog of the source, it looks like kubebuilders limits are per-reconciler, rather than per-manager. can you confirm this?

Reply:
Yep, that's correct 😄 you can also have N 'worker' goroutines for each reconciler too - I guess we'll set this to `1` for this particular reconciler 😄